webserver/views/api/v1/core.py

```diff
@@ -248,14 +256,25 @@ def get_many_highlevel():
     in the returned data.
 
     :query recording_ids: *Required.* A list of recording MBIDs to retrieve
+    :query map_model_class_names: *Optional.* Boolean flag to get descriptive keys
 
       Takes the form `mbid[:offset];mbid[:offset]`. Offsets are optional, and should
       be >= 0
 
     :resheader Content-Type: *application/json*
     """
+
     recording_details = get_data_for_multiple_recordings(db.data.load_high_level)
-    return recording_details
+    map_keys = True if request.args.get(
+        "map_model_class_names") == 'True' else False
+    if map_keys:
+        for recording_id, offset in recording_details.iteritems():
+            for key, data in offset.iteritems():
+                recording_details.setdefault(recording_id, {})[key] = db.data.map_class_labels(
```

Comment:
Is there a reason why you use `setdefault` here? can you not just use `recording_details[recording_id][key]`?